app/views/repositories/_repository_header.html.erb

```diff
@@ -28,12 +28,49 @@ See COPYRIGHT and LICENSE files for more details.
 ++#%>
 
 <opce-persistent-toggle data-identifier="repository.checkout_instructions"></opce-persistent-toggle>
-<%= toolbar title: t(
+
+<%=
+  render Primer::OpenProject::PageHeader.new do |header|
+    header.with_title { t(
       "repositories.named_repository",
       vendor_name: @repository.class.vendor_name
-    ) do %>
-  <% if @instructions && @instructions.available? %>
-  <li class="toolbar-item toolbar-input-group hidden-for-mobile">
+    ) }
+    header.with_breadcrumbs(breadcrumbs(project: @project, path: @path, rev: @rev))
+    if !empty && User.current.allowed_in_project?(:browse_repository, @project)
+      header.with_action_icon_button(
+        tag: :a,
+        icon: :graph,
+        label: t(:label_statistics),
+        mobile_icon: :graph,
+        mobile_label: t(:label_statistics),
+        size: :medium,
+        href: stats_project_repository_path(@project),
+        aria: { label: t(:label_statistics) },
+        title: t(:label_statistics)
+      )
+    end
+    if User.current.allowed_in_project?(:manage_repository, @project)
+      header.with_action_icon_button(
+        tag: :a,
+        icon: :gear,
+        label: t(:label_setting_plural),
+        mobile_icon: :gear,
+        mobile_label: t(:label_setting_plural),
+        size: :medium,
+        href: project_settings_repository_path(@project),
+        aria: { label: t(:label_setting_plural) },
+        title: t(:label_setting_plural)
+      )
+    end
+  end
+%>
+<%= call_hook(
+      :repositories_navigation_toolbar,
+      { repository: @repository, project: @project, repository_empty: empty }
+    ) %>
```

Comment:
This will not work any more. I guess, we have to find out whether this is used by a customer, otherwise remove it.